submit/submit

```diff
@@ -0,0 +1,534 @@
+    magic = None
+
+# By default no base URL is set and it is expected to use either the
+# SUBMITBASEURL environment variable or the -u/--url argument.
+# If you want to hardcode a default base URL, set it here
+baseurl = ''
+
+# Use a specific API version, set to empty string for default
+api_version = 'v4/'
+
+# Last modified time in minutes after which to warn for submitting
+# an old file.
+warn_mtime = 5
+
+num_warnings = 0
+
+def confirm(message: str) -> bool:
+    answer = ''
+    while answer not in ['y', 'n']:
+        answer = input(' (y/n) '.format(message)).lower()
+    return answer == 'y'
+
+def warn_user(msg: str):
+    global num_warnings
+    if args.quiet:
+        logging.debug('user warning #{}: {}'.format(num_warnings, msg))
+    else:
+        print('WARNING: {}!'.format(msg))
+    num_warnings += 1
+
+def usage(msg: str):
+    logging.error('error: {}'.format(msg))
+
+    print("Type '{} --help' to get help.".format(sys.argv[0]))
+
+    exit(1)
+
+def error(msg: str):
+    logging.error(msg)
+    exit(-1)
+
+def read_contests() -> list:
+    '''Read all contests from the API
+    
+    Returns:
+        The contests or None if an error occurred
+    '''
+
+    try:
+        data = do_api_request('contests')
+    except RuntimeError as e:
+        logging.warning(e)
+        return None
+
+    if not isinstance(data, list):
+        logging.warning('REST API returned unexpected JSON data for endpoint contests')
+        return None
+
+    contests = []
+
+    for contest in data:
+        if ('id' not in contest or 'shortname' not in contest or contest['id'] == ''
+            or contest['shortname'] == ''):
+            logging.warning('REST API returned unexpected JSON data for contests')
+            return None
+        contests.append(contest)
+
+    logging.info('read {} contests from the API'.format(len(contests)))
+
+    return contests
+
+def read_languages() -> list:
+    '''Read all languages for the current contest from the API
+    
+    Returns:
+        The languages or None if an error occurred
+    '''
+
+    try:
+        endpoint = 'contests/' + my_contest['id'] + '/languages'
+        data = do_api_request(endpoint)
+    except RuntimeError as e:
+        logging.warning(e)
+        return None
+
+    if not isinstance(data, list):
+        logging.warning('REST API returned unexpected JSON data for endpoint languages')
+        return None
+
+    languages = []
+
+    for item in data:
+        if ('id' not in item or 'extensions' not in item
+            or item['id'] == '' or not isinstance(item['extensions'], list)
+            or len(item['extensions']) == 0):
+            logging.warning('REST API returned unexpected JSON data for languages')
+            return None
+        language = {
+            'id': item['id'],
+            'name': item['name'],
+            'entry_point_required': item['entry_point_required'] or False,
+            'extensions': {item['id']}
```

Comment:
Just on the phone, so can't check, but does it make sense to have a dict here instead of a list.?

Reply:
It's a set right, not a dict? At least that is what I found about `{item}`. I do this to make sure the items are unique (since we insert both the ID and extension). Can change it to a list but then we need to make it unique.

Reply:
Ah, ok, yeah that works